apps/greencheck/tests/test_azure_cloud_importer.py

```diff
@@ -0,0 +1,114 @@
+import pytest
+import pathlib
+import ipaddress
+import json
+from io import StringIO
+
+from django.core.management import call_command
+from apps.greencheck.management.commands import update_azure_ip_ranges
+from apps.accounts.models import Hostingprovider
+from apps.greencheck.models import GreencheckIp
+from apps.greencheck.management.commands.update_azure_ip_ranges import GREEN_REGIONS
+
+from django.conf import settings
+
+
+@pytest.fixture
+def hosting_provider():
+
+    # oregon, *rest = [region for region in GREEN_REGIONS if region[1] == "us-west-2"]
+    name, region, host_id = ("Azure US West", "az-west-2", 123)
+    return Hostingprovider(
+        archived=False,
+        country="US",
+        customer=False,
+        icon="",
+        iconurl="",
+        id=host_id,
+        model="groeneenergie",
+        name=name,
+        partner="",
+        showonwebsite=True,
+        website="http://azure.microsoft.com",
+    )
+
+
+@pytest.fixture
+def azure_cloud_provider(hosting_provider):
+    hosting_provider.save()
+    return update_azure_ip_ranges.MicrosoftCloudProvider()
+
+
+@pytest.fixture
+def azure_json_ip_ranges():
+    this_file = pathlib.Path(__file__)
```

Comment:
Nice! Thanks you for using pathlib here! 💚  this is a good norm to apply for all access of local files to read